### DISCUSSION TITLE
When DataGrid is not set to a height, dynamically adding rows will not widen the height

### DISCUSSION THREAD

**Discussion by mfkwgizhoujian** [AvaloniaUI/Avalonia#16022] [Q&A]
### Describe the bug

When I was using DataGrid, I did not set the height, and the set for initializing the binding was empty. By clicking the button to add rows, there was data in the set, but it did not expand the height of the DataGrid, which led me to think that there was a problem with the binding at one point.

<img width="1020" alt="image" src="https://github.com/AvaloniaUI/Avalonia/assets/33543763/d00b1940-c05f-43a7-9e06-2a76c6af219e">

### To Reproduce

1. Create a DataGrid normally
2. Add a button
3. Add a click event to the button and add data to the ObservableCollection object bound to the DataGrid through the Add method

### Expected behavior

When dynamically adding data, if the height of the DataGrid is not set, it should automatically expand to display the latest data

### Avalonia version

11.0.10

### OS

Windows, macOS

**Comment by workgroupengineering** [comment]
```xaml
 <StackPanel>
     <Button Content="Add"/>
     <DataGrid  ScrollViewer.VerticalScrollBarVisibility="Disabled">
     </DataGrid> 
 </StackPanel>
```

For next , open the discussion before open the issue.

**Comment by mfkwgizhoujian** [comment]
<img width="988" alt="image" src="https://github.com/AvaloniaUI/Avalonia/assets/33543763/720147e7-8368-414e-94ed-27e64ab62137">

https://github.com/AvaloniaUI/Avalonia/assets/33543763/069cfaca-e806-4a07-99b6-d2c2e43929f6

I added this code: ScrollViewer.VerticalScrollBarVisibility="Disabled", only adding the first piece of data can automatically display the data, and the rest cannot be extended unless scrolling the mouse over the table

**Comment by workgroupengineering** [comment]
Maybe there's a problem with your theme. With the fluent theme it seems to work.

eg: [TAGrid.zip](https://github.com/user-attachments/files/15820674/TAGrid.zip)

**Comment by timunie** [comment]
DataGrid needs fixed height to work as inteneded. StackPanel doesn't provide it. Either set height on your own or use a Grid or DockPanel.
